File: tests/expectations/both/bitflags.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

/**
 * Constants shared by multiple CSS Box Alignment properties
 * These constants match Gecko's `NS_STYLE_ALIGN_*` constants.
 */
typedef struct AlignFlags {
  uint8_t bits;
} AlignFlags;
#define AlignFlags_AUTO (AlignFlags){ .bits = 0 }
#define AlignFlags_NORMAL (AlignFlags){ .bits = 1 }
#define AlignFlags_START (AlignFlags){ .bits = 1 << 1 }
#define AlignFlags_END (AlignFlags){ .bits = 1 << 2 }
#define AlignFlags_FLEX_START (AlignFlags){ .bits = 1 << 3 }

void root(AlignFlags flags);

File: tests/expectations/bitflags.zig
const std = @import("std");

pub const AlignFlags = extern struct {
    bits: u8,
};

pub const DebugFlags = extern struct {
    bits: u32,
};

extern fn root(flags: AlignFlags, bigger_flags: DebugFlags) void;

pub const AlignFlags_AUTO = std.mem.zeroInit(AlignFlags, .{
    .bits = std.zig.c_translation.cast(u8, @as(c_int, 0)),
});

pub const AlignFlags_NORMAL = std.mem.zeroInit(AlignFlags, .{
    .bits = std.zig.c_translation.cast(u8, @as(c_int, 1)),
});

pub const AlignFlags_START = std.mem.zeroInit(AlignFlags, .{
    .bits = std.zig.c_translation.cast(u8, @as(c_int, 1) << @as(c_int, 1)),
});

pub const AlignFlags_END = std.mem.zeroInit(AlignFlags, .{
    .bits = std.zig.c_translation.cast(u8, @as(c_int, 1) << @as(c_int, 2)),
});

pub const AlignFlags_FLEX_START = std.mem.zeroInit(AlignFlags, .{
    .bits = std.zig.c_translation.cast(u8, @as(c_int, 1) << @as(c_int, 3)),
});

pub const DebugFlags_BIGGEST_ALLOWED = std.mem.zeroInit(DebugFlags, .{
    .bits = std.zig.c_translation.cast(u32, @as(c_int, 1) << @as(c_int, 31)),
});
